vc_zoom/indico_vc_zoom/forms.py

```diff
@@ -0,0 +1,110 @@
+                                    choices=[
+                                        ('regular', _('Regular Meeting')),
+                                        ('webinar', _('Webinar'))])
+
+    host_choice = IndicoRadioField(_("Meeting Host"), [DataRequired()],
+                                   choices=[('myself', _("Myself")), ('someone_else', _("Someone else"))])
+
+    host_user = PrincipalField(_("User"),
+                               [HiddenUnless('host_choice', 'someone_else'), DataRequired()])
+
+    password = StringField(_("Passcode"),
+                           [DataRequired(), IndicoRegexp(r'^\d{8,}$')],
+                           description=_("Meeting passcode (min. 8 digits)"))
+
+    password_visibility = IndicoRadioField(_("Passcode visibility"),
+                                           description=_("Who should be able to know this meeting's passcode"),
+                                           orientation='horizontal',
+                                           choices=[
+                                               ('everyone', _('Everyone')),
+                                               ('logged_in', _('Logged-in users')),
+                                               ('no_one', _("No one"))])
+
+    mute_audio = BooleanField(_('Mute audio'),
+                              widget=SwitchWidget(),
+                              description=_('Participants will join the VC room muted by default '))
+
+    mute_host_video = BooleanField(_('Mute video (host)'),
+                                   widget=SwitchWidget(),
+                                   description=_('The host will join the VC room with video disabled'))
+
+    mute_participant_video = BooleanField(_('Mute video (participants)'),
+                                          widget=SwitchWidget(),
+                                          description=_('Participants will join the VC room with video disabled'))
+
+    waiting_room = BooleanField(_('Waiting room'),
+                                widget=SwitchWidget(),
+                                description=_('Participants may be kept in a waiting room by the host'))
+
+    description = TextAreaField(_('Description'), description=_('Optional description for this room'))
+
+    def __init__(self, *args, **kwargs):
+        defaults = kwargs['obj']
+        if defaults.host_user is None and defaults.host is not None:
+            host = principal_from_identifier(defaults.host)
+            defaults.host_choice = 'myself' if host == session.user else 'someone_else'
+            defaults.host_user = None if host == session.user else host
+        super(VCRoomForm, self).__init__(*args, **kwargs)
+
+        if defaults.meeting_type is None:
+            del self.meeting_type
+        else:
+            for f in {self.mute_audio, self.mute_participant_video, self.waiting_room}:
+                f.validators = [HiddenUnless('meeting_type', 'regular')]
```

Comment:
Using our `inject_validators` util here may be safer (note that this should be done before the super call)